extensions/2.0/Khronos/KHR_interactivity/Specification.adoc

```diff
@@ -687,30 +860,59 @@ If any input value is _NaN_, the output value is false.
 
 [cols="1h,1,2"]
 |===
-| Type | `math/select` | Conditional selection operation
+| Operation | `math/select` | Conditional selection operation
 .3+| Input value sockets
 | `bool condition` | Value selecting the value returned
 | `T a` | Positive selection option
 | `T b` | Negative selection option
 | Output value sockets
-| `T value` | stem:[a] if the the `condition` input value is true; stem:[b] otherwise
+| `T value` | stem:[a] if the `condition` input value is true; stem:[b] otherwise
 |===
 
-The type `T` represents any type. It **MUST** be the same for the output value socket and the input value sockets stem:[a] and stem:[b], otherwise the node is invalid.
+The type `T` represents any supported type including custom types. It **MUST** be the same for the output value socket and the input value sockets stem:[a] and stem:[b].
+
+===== Random
+
+[cols="1h,1,2"]
+|===
+| Operation | `math/random` | Random value generation operation
+| Output value sockets | `float value` | A pseudo-random number greater than or equal to zero and less than one
+|===
+
+[WARNING]
+====
+This node is not intended for any workflows that require cryptographically secure random numbers.
+====
+
+The value of the output value socket `value` **MUST** be initialized to a random number on the first access. Any two accesses of the output value socket `value` **MUST** return the same value if there were no flow socket activations (of other nodes) between them.
+
+[NOTE]
+.Example
+====
+This means that, e.g., a `math/eq` node with both its input value sockets connected to the same `math/random` node always returns true.
+====
+
+The value of the output value socket `value` **SHOULD** be updated when accessed as a result of a new flow socket activation, including self-activations.
```

Comment:
Why is this a SHOULD and not a MUST? Feels weird that this is undefined behaviour. If the goal was to say "the new value can be the same as the old value" that could be added instead.

Reply:
SHOULD is not undefined, its normative definition from RFC 2119 is:
> This word, or the adjective "RECOMMENDED", mean that there may exist valid reasons in particular circumstances to ignore a particular item, but the full implications must be understood and carefully weighed before choosing a different course.

That said, I think it's safe to use MUST here.

Reply:
Thanks! I didn't mean to say that "should" is undefined, just that graph authors can't rely on the behaviour of `math/random` if it only "should" be updated in the case described here.